assignment_1/2_kth_element.py

```diff
@@ -0,0 +1,101 @@
+# Implement an algorithm to find the kth to last element of a singly linked list
+# Magdalena Grodzińska
+
+# linked list implementation
+
+class Node:
+    def __init__(self, data, next_node):
+        self.data = data
+        self.next_node = next_node
+
+    def get_data(self):
+        return self.data
+
+    def set_data(self):
+        self.data = new_data
+
+    def get_next(self):
+        return self.next_node
+
+    def set_next(self, new_next):
+        self.next_node = new_next
+
+
+# main function
+
+def find_kth(list, k):
+    if list == None:
+        return None
+
+    length = count_elements(list)
+
+    if k >= length:
+        return None;
+
+    how_many_to_go = length - k
+    return iterate_n_elements(list, how_many_to_go)
+
+
+# helper functions
+
+def count_elements(list):
+    counter = 0
+    node_i = list
+
+    while node_i != None:
+        counter += 1
+        node_i = node_i.get_next()
+
+    return counter
+
+
+def iterate_n_elements(list, n):
+    counter = 0
+    node_i = list
+    element = None
+
+    while counter < n:
+        element = node_i.get_data()
+        #print(str(counter) + " " + str(element))
+        node_i = node_i.get_next()
+        counter += 1
+
+    return element
+
+
+# testing functions
+
+def build_sample_list(n):
```

Comment:
The Node() class is usable as a linked list, but requiring this helper for test cases suggests that the linked list implementation could use a convenient way to construct linked lists as well.

It's a "separation of concerns" and/or abstractions question - it makes your code easier to understand and maintain if functions dealing with details about linked lists implementation are together in one place and the code using linked lists can do so in a straightforward way.